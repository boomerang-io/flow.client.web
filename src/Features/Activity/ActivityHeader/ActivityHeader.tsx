//@ts-nocheck
import React from "react";
import PropTypes from "prop-types";
import HeaderWidget from "Components/HeaderWidget";
import {
  FeatureHeader as Header,
  FeatureHeaderTitle as HeaderTitle,
  FeatureHeaderSubtitle as HeaderSubtitle,
} from "@boomerang-io/carbon-addons-boomerang-react";
import { SkeletonPlaceholder, Breadcrumb, BreadcrumbItem } from "@carbon/react";
import { ArrowDownRight, ArrowUpRight } from "@carbon/react/icons";
import { Link } from "react-router-dom";
import { appLink } from "Config/appConfig";
import styles from "./activityHeader.module.scss";

ActivityHeader.propTypes = {
  inProgressActivities: PropTypes.number,
  isError: PropTypes.bool,
  isLoading: PropTypes.bool.isRequired,
  failedActivities: PropTypes.number,
  runActivities: PropTypes.number,
  team: PropTypes.any,
  succeededActivities: PropTypes.number,
};

function ActivityHeader({
  inProgressActivities,
  isError,
  isLoading,
  runActivities,
  team,
  succeededActivities,
  failedActivities,
}) {
  const successRate = runActivities > 0 ? (succeededActivities + inProgressActivities) / runActivities : 0;
  const successRatePercentage = Math.round(successRate * 100);
  const emoji = successRatePercentage > 79 ? "ðŸ™Œ" : successRatePercentage > 49 ? "ðŸ˜®" : "ðŸ˜¨";

  const NavigationComponent = () => {
    return (
      <Breadcrumb noTrailingSlash>
        <BreadcrumbItem>
          <Link to={appLink.home()}>Home</Link>
        </BreadcrumbItem>
        <BreadcrumbItem isCurrentPage>
          <p>{team.displayName}</p>
        </BreadcrumbItem>
      </Breadcrumb>
    );
  };

  return (
    <Header
      className={styles.container}
      includeBorder={false}
      nav={<NavigationComponent />}
      header={
        <>
          <HeaderTitle>Activity</HeaderTitle>
          <HeaderSubtitle className={styles.headerMessage}>The place to view WorkflowRuns</HeaderSubtitle>
        </>
      }
      actions={
        <section className={styles.content}>
          {isLoading ? (
            <SkeletonPlaceholder className={styles.summarySkeleton} />
          ) : isError ? (
            <>
              <p className={styles.text}>Today's numbers</p>
              <HeaderWidget text="Runs" value={"--"} />
              <HeaderWidget text="In Progress" value={"--"} />
              <HeaderWidget text="Successes" value={"--"} />
              <HeaderWidget text="Failures" value={"--"} />
              <HeaderWidget text="Success rate" value={"--"} />
            </>
          ) : (
            <>
              <p className={styles.text}>Today's numbers</p>
              <HeaderWidget icon={ArrowUpRight} text="Runs" value={runActivities} />
              <HeaderWidget icon={ArrowUpRight} text="In Progress" value={inProgressActivities} />
              <HeaderWidget icon={ArrowUpRight} text="Successes" value={succeededActivities} />
              <HeaderWidget icon={ArrowDownRight} text="Failures" value={failedActivities} />
              <HeaderWidget icon={emoji} text="Success rate" value={`${successRatePercentage}%`} />
            </>
          )}
        </section>
      }
    />
  );
}

export default ActivityHeader;
