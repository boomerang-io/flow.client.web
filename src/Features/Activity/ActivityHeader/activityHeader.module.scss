.container {
  h1 {
    margin: 0;
  }

  h2 {
    font-size: 1.25rem;
    font-weight: 300;
  }
}

.content {
  align-items: center;
  display: flex;
  margin: 0.5rem 2.125rem 1.5625rem 0;
}

.text {
  color: var(--cds-text-01);
  font-size: 1.25rem;
  font-weight: 300;
  line-height: 1.5625rem;
}

.summarySkeleton {
  width: 25rem;
  height: 4rem;
}

.headerMessage {
  margin-top: 0.5rem;
}
