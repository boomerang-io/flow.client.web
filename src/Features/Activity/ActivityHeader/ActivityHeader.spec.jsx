import React from "react";
import ActivityHeader from "./index";

const props = {
  failedActivities: 10,
  runActivities: 25,
  succeededActivities: 13,
  inProgressActivities: 0,
  isLoading: false,
};

describe("ActivityHeader --- Snapshot", () => {
  it("Capturing Snapshot of ActivityHeader", () => {
    const { baseElement } = rtlRender(<ActivityHeader {...props} />);
    expect(baseElement).toMatchSnapshot();
  });
});
