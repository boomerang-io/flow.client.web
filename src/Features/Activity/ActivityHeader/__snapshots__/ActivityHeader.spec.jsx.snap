// Vitest Snapshot v1

exports[`ActivityHeader --- Snapshot > Capturing Snapshot of ActivityHeader 1`] = `
<body
  id="app"
>
  <div>
    <header
      class="bx--bmrg-feature-header _container_76bf80"
    >
      <section
        class="bx--bmrg-feature-header__content"
      >
        <div
          class="bx--bmrg-feature-header__header"
        >
          <p
            class="bx--bmrg-feature-header-title"
          >
            This is all of the
          </p>
          <h1
            class="bx--bmrg-feature-header-title"
          >
            Activity
          </h1>
        </div>
      </section>
      <section
        class="_content_76bf80"
      >
        <p
          class="_text_76bf80"
        >
          Today's numbers
        </p>
        <div
          class="_container_87f8c4"
        >
          <div
            class="_valueAndIcon_87f8c4"
          >
            <p
              class="_value_87f8c4"
            >
              25
            </p>
            <svg
              aria-label="Runs"
              class="_icon_87f8c4"
              fill="currentColor"
              focusable="false"
              height="32"
              preserveAspectRatio="xMidYMid meet"
              role="img"
              viewBox="0 0 32 32"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 6L10 8 22.59 8 6 24.59 7.41 26 24 9.41 24 22 26 22 26 6 10 6z"
              />
            </svg>
          </div>
          <p
            class="_text_87f8c4"
          >
            Runs
          </p>
        </div>
        <div
          class="_container_87f8c4"
        >
          <div
            class="_valueAndIcon_87f8c4"
          >
            <p
              class="_value_87f8c4"
            >
              13
            </p>
            <svg
              aria-label="Successes"
              class="_icon_87f8c4"
              fill="currentColor"
              focusable="false"
              height="32"
              preserveAspectRatio="xMidYMid meet"
              role="img"
              viewBox="0 0 32 32"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 6L10 8 22.59 8 6 24.59 7.41 26 24 9.41 24 22 26 22 26 6 10 6z"
              />
            </svg>
          </div>
          <p
            class="_text_87f8c4"
          >
            Successes
          </p>
        </div>
        <div
          class="_container_87f8c4"
        >
          <div
            class="_valueAndIcon_87f8c4"
          >
            <p
              class="_value_87f8c4"
            >
              10
            </p>
            <svg
              aria-label="Failures"
              class="_icon_87f8c4"
              fill="currentColor"
              focusable="false"
              height="32"
              preserveAspectRatio="xMidYMid meet"
              role="img"
              viewBox="0 0 32 32"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 26L10 24 22.59 24 6 7.41 7.41 6 24 22.59 24 10 26 10 26 26 10 26z"
              />
            </svg>
          </div>
          <p
            class="_text_87f8c4"
          >
            Failures
          </p>
        </div>
        <div
          class="_container_87f8c4"
        >
          <div
            class="_valueAndIcon_87f8c4"
          >
            <p
              class="_value_87f8c4"
            >
              52%
            </p>
            <span
              aria-hidden="false"
              aria-label="Success rate"
              class="_emoji_87f8c4"
              role="img"
            >
              ðŸ˜®
            </span>
          </div>
          <p
            class="_text_87f8c4"
          >
            Success rate
          </p>
        </div>
      </section>
    </header>
  </div>
</body>
`;
